### PR TITLE
86 add monitor

### PR DESCRIPTION
**Pull request by snorkellingcactus**
In theory it works by dragging one or more property (a dynamic readable one) to the monitor widget

Althrough selection does not work as ideal
I messed up the layout by inserting the monitor widget
I weren't able to get double values from the dbus interface

### CODE REVIEW

File: src/tuxclocker-qt/monitor/Monitor.cpp [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+void Monitor::setupLessImportant(){
+  QSharedPointer<QCPAxisTickerTime> timeTicker(new QCPAxisTickerTime);
+  timeTicker->setTimeFormat("%m:%s:%z");
+  timeTicker->setTickCount(8);
+  xAxis->setTicker(timeTicker);
+
+  xAxis->setTickLabelFont(QFont(QFont().family(), 12));
+  yAxis->setTickLabelFont(QFont(QFont().family(), 12));
+  xAxis->setTickLabelColor(QColor(200, 200, 200));
```
Comment (Lurkki14) [inline src/tuxclocker-qt/monitor/Monitor.cpp:65]: Colors shouldn't be hardcoded https://doc.qt.io/qt-6/qpalette.html#color-1

**Review by Lurkki14 — CHANGES_REQUESTED** [review]

File: src/tuxclocker-qt/meson.build [hunk 2]
```diff
@@ -9,6 +9,8 @@ moc_files = qt5.preprocess(moc_headers: ['MainWindow.hpp',
 		'data/AssignableItem.hpp',
 		'data/AssignableProxy.hpp',
 		'data/DeviceModel.hpp',
+		'monitor/Monitor.hpp',
+		'monitor/qcustomplot.h',
```
[the remark this answers is not in the document]
Comment (snorkellingcactus) [reply src/tuxclocker-qt/meson.build:13]: No special reason, i can try to use Qt charts. I've just started with this and then discovered your project and qt charts. I can compare them after doing a qt chart version.

File: src/tuxclocker-qt/monitor/Metric.cpp [hunk 3]
```diff
@@ -0,0 +1,92 @@
+#include "Metric.hpp"
```
[the remark this answers is not in the document]
Comment (snorkellingcactus) [reply src/tuxclocker-qt/monitor/Metric.cpp:1]: I'll try that

File: src/tuxclocker-qt/monitor/Monitor.cpp [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+  textTicker->addTick(0.4 * size, "40%");
+  textTicker->addTick(0.6 * size, "60%");
+  textTicker->addTick(0.8 * size, "80%");
+  textTicker->addTick(size, "100%");
+}
+
+
+void Monitor::setupLessImportant(){
+  QSharedPointer<QCPAxisTickerTime> timeTicker(new QCPAxisTickerTime);
+  timeTicker->setTimeFormat("%m:%s:%z");
```
[the remark this answers is not in the document]
Comment (snorkellingcactus) [reply src/tuxclocker-qt/monitor/Monitor.cpp:58]: Sure

**Pull request closed without merging**